src/IrcTransport/IrcWebSocketTransport.cs

```diff
@@ -0,0 +1,348 @@
+﻿/*
+ * $Id$
+ * $URL$
+ * $Rev$
+ * $Author$
+ * $Date$
+ *
+ * SmartIrc4net - the IRC library for .NET/C# <http://smartirc4net.sf.net>
+ *
+ * Copyright (c) 2015 Katy Coe <djkaty@start.no> <http://www.djkaty.com>
+ * 
+ * Full LGPL License: <http://www.gnu.org/licenses/lgpl.txt>
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using WebSocketSharp;
+using WebSocketSharp.Net;
+
+namespace Meebey.SmartIrc4net
+{
+    /// <summary>
+    /// WebSocket-based IRC protocol using websocket-sharp library
+    /// </summary>
+    public class IrcWebSocketTransport : IIrcTransportManager
+    {
+        private bool _IsConnected;
+        private bool _IsConnectionError;
+        private bool _IsDisconnecting;
+        private bool _IsConnecting;
+
+        /// <summary>
+        /// Event which fires when a line of text is received from the connection
+        /// </summary>
+        public event ReadLineEventHandler OnMessageReceived;
+
+        /// <summary>
+        /// Event which fires when a connection error occurs
+        /// </summary>
+        public event Action OnConnectionError;
+
+        /// <summary>
+        /// The underlying WebSocket of the connection
+        /// </summary>
+        [CLSCompliant(false)]
+        public WebSocket Socket { get; set; }
```

Comment:
Originally yes, so the user could alter the WebSocket configuration if required. Now all of the properties are set in IrcWebSocketTransport via proxy properties. Do you want me to change it to private?